gridfinity_drawers.scad

```diff
@@ -0,0 +1,126 @@
+use <modules/gridfinity_cup_modules.scad>
+use <gridfinity_baseplate.scad>
+//From https://www.printables.com/pl/model/363389-gridfinity-drawer-chest-remix
+
+width = 4;
+depth = 3;
+height = 4;
+count = 3;
+//heightsRaw = "4 4 4";
+clearance = 0.25;
+wallthicknessInner = 2;
+wallthicknessOuter = 2;
+handlewidth = 40;
+handleheight = 4;
+handlelength = 7;
+ridgedepth = 5;
+ridgethickness = 1;
+bottomgrid = true;
+topgrid = true;
+mode = "everything"; //["everything", "drawers", "holder", "onedrawer"]
+
+//heights = str_split(heightsRaw, " ");
+InnerDrawerW = (width*42) + clearance - 0.25;
+InnerDrawerD = (depth*42) + clearance - 0.25;
+InnerDrawerH = (height*7) + clearance + 4.25;
+OuterDrawerW = InnerDrawerW + (wallthicknessInner * 2);
+OuterDrawerD = InnerDrawerD + (wallthicknessInner * 2);
+OuterDrawerH = InnerDrawerH + wallthicknessInner;
+InnerBoxW = OuterDrawerW + (clearance * 2);
+InnerBoxD = OuterDrawerD + (clearance * 2);
+InnerBoxH = OuterDrawerH + (clearance * 2);
+OuterBoxW = InnerBoxW + (wallthicknessOuter * 2);
+OuterBoxD = InnerBoxD + (wallthicknessOuter);
+BottomGridOffset = wallthicknessInner + wallthicknessOuter + clearance*2;
+TopGridOffset = BottomGridOffset - 0.25;
+
+HoleH = OuterDrawerH + (clearance * 2);
+TotalH = (HoleH * count) + (ridgethickness * (count - 1)) + (wallthicknessOuter * 2);
+IncrementH = HoleH + ridgethickness;
+StartH = wallthicknessOuter;
+OffsetW = wallthicknessOuter + clearance;
+
+
+//DRAWER STUFF
+module rounddrawerbox(w, d, h, r=6){
+    D = r * 2;
+    linear_extrude(height = h)
+    minkowski(){
+        translate([r, r]) square([w-D, d-D]);
+        circle(r);
+    }
+}
+module drawer(h){
+    InnerH = (h*7) + clearance + 4.25;
+    OuterH = InnerH + wallthicknessInner;
+    union(){
+        difference(){
+            rounddrawerbox(OuterDrawerW, OuterDrawerD, OuterH);
+            drawerCutout(h);
+        };
+        translate([OuterDrawerW / 2, 0, OuterH / 2]) handle();
+    }
+    
+}
+module handle(){
+    translate(-[handlewidth / 2, handlelength, handleheight/2]) cube([handlewidth, handlelength, handleheight]);
+}
+module drawerCutout(h){
+    translate([wallthicknessInner, wallthicknessInner, wallthicknessInner]) difference(){
+        rounddrawerbox(InnerDrawerW, InnerDrawerD, 99999, 4);
+        translate([21, 21, 0]) frame_plain(width, depth);
+    }
+}
+module drawers(){
+    for(i = [0 : count-1]){
+        vpos = clearance + StartH + IncrementH * i;
+        color("red") translate([0, OffsetW, vpos]) drawer(height);
+    }
+}
+
+//HOLDER STUFF
+module baseUnit(){
+    import("Bin Base - Printables model 417152.stl");
+}
```

Comment:
@ostat How does the grid block work? In my tests, using it in the union did not add it to the model.

Reply:
I am creating a non stackable bin (meaning there is no lip around the top) in the required dimensions. Dropping the height so that it looked like it matched what you had setup.

It seem to be working OK from my testing. If there are issues with height or a gap, we can adjust the height to make it work.